### PR TITLE
Wallet add custom RPC endpoint

### PR DESCRIPTION
**Pull request by Jibz1**
- Add network option for RPC endpoint 
- Add network option for Testnet 
- Feature flag with Growthbook
- Small nonrelated fix. fix error text-overflow on the activities section

https://user-images.githubusercontent.com/8676844/199635650-f45df962-8163-438f-81bd-e82a455f9e52.mov

https://user-images.githubusercontent.com/8676844/199800499-5d49d185-a048-4951-96ad-defeafe0df38.mov

https://user-images.githubusercontent.com/8676844/199825100-a33d7ecc-659c-41e6-9a5d-d81b6cb46601.mov

### CODE REVIEW

File: apps/wallet/src/ui/app/components/network-selector/index.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/network-selector/index.tsx:65]: This shouldn't be a link. It should be a `button` with `type="button"`.

Comment (Jibz1) [reply apps/wallet/src/ui/app/components/network-selector/index.tsx]: done

File: apps/wallet/src/ui/app/components/network-selector/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/network-selector/index.tsx]: done

**Review by mystie711 — CHANGES_REQUESTED** [review]
Copy updates needed
- Sui Devnet
- Sui Testnet

**Review by mystie711 — APPROVED** [review]

**Comment by Jibz1** [comment]
@mystie711 please see update.

File: apps/wallet/src/ui/app/components/network-selector/index.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/network-selector/index.tsx:34]: Just make this a boolean instead of evaluated into an `API_ENV` value. This approach is more confusing than it's worth.

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:21]: Should we validate this is a URL?

File: apps/wallet/src/ui/app/components/explorer-link/Explorer.ts
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:14]: localhost:3000?

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:21]: Not sure how to correctly validate it, if we factor localhost?

Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:21]: Could just ensure the URL starts with `http://` or `https://`

File: apps/wallet/src/ui/app/components/explorer-link/Explorer.ts
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:14]: I don't think we should include the explorer option unless we provide an input field for that. It will be confusing if the explorer link is NOT on `localhost:3000`

Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:14]: Yeah that's fair, I'm okay with that.

File: apps/wallet/src/ui/app/components/network-selector/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/network-selector/index.tsx:34]: Oh, I should change the label `activeClass`, which refers to the selected/active network name, which may differ from the saved network name.

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:38]: Do we really need to keep a local state of the custom rpc url? Why not just use the one from the store? `setCustomRPC` action will update it right?

File: apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx:27]: Fragment is not necessary right?

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:38]: we do not save to store until a valid RPC endpoint has been added. Plus, we need it for visual purpose.

File: apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx
Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx:27]: I thought that react does not allow the return of empty/null component.

File: apps/wallet/src/ui/app/components/network-selector/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/network-selector/index.tsx:45]: this one seems similar to `generateActiveNetworkList` maybe we should use that?

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:25]: why not use https://github.com/jquense/yup#stringurlmessage-string--function-schema

File: apps/wallet/src/ui/app/components/explorer-link/Explorer.ts
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:28]: why do why need `?` here?

File: apps/wallet/src/ui/app/components/explorer-link/Explorer.ts
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:42]: Same as above

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:35]: This is always empty right?

Comment (Jibz1) [reply apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:35]: You are right. Previously I had a default URL for custom RPC. I will remove it

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:38]: I am not sure I get your point. `customRPCURL` seems to be used only as an initial value for the form. If the user changes the input it uses the internal state of the form and when the user clicks save we update the url in store and here. So we could change 

```
initialValues={{ rpcInput: customRPCURL }}
```
to 
```
initialValues={{ rpcInput: customRPC || '' }}
```
and remove the local state

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:25]: Did you try it? I found [this](https://github.com/jquense/yup/pull/844) so it might not

File: apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx:29]: We could use `new URL()` with try catch and return false on error. `http(s)://` is not a valid url either

File: apps/wallet/src/ui/app/components/explorer-link/Explorer.ts
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:28]: Just missed it.

Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:28]: I don't think it's necessary though. new URL either will through or be a URL

File: apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx:27]: null should be fine

Comment (Jibz1) [reply apps/wallet/src/ui/app/shared/faucet/request-button/index.tsx:27]: Nope. does not work

File: apps/wallet/src/ui/app/components/explorer-link/Explorer.ts
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/components/explorer-link/Explorer.ts:28]: Screenshot 2022-11-08 at 15 08 35

**Review by pchrysochoidis — APPROVED** [review]
lgtm - ideally wait for @Jordan-Mysten to approve in case he has any comments that are not addressed before merging

File: apps/wallet/src/ui/app/ApiProvider.ts
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/ApiProvider.ts:93]: I don't love this `&&` pattern, I'd just keep these as booleans and push in the `API_ENV` value in the if body.

**Review by Jordan-Mysten — APPROVED** [review]
Approving to unblock

File: apps/wallet/src/ui/app/ApiProvider.ts
Comment (Jibz1) [reply apps/wallet/src/ui/app/ApiProvider.ts]: done.

**Pull request merged**
